CONTRIBUTING.md

```diff
@@ -0,0 +1,123 @@
+
+	    composer install
+
+    This will automatically install the `pre-commit` hook from the `wp-dev-lib` Composer package and setup the unit tests.
+
+### Docker
+
+This repository includes a WordPress development environment based on [Docker](https://docs.docker.com/install/) that can be run on your computer.
+
+To use the Docker based environment with the Docker engine running on your host, run:
+
+	docker-compose up
+
+which will make it available at [localhost](http://localhost). Ensure that no other Docker containers or services are using port `80` or `3306` on your machine. 
+
+**Important**: You must execute the `docker-compose up` command before the `pre-commit` hook will work properly. This is because the unit tests depend on the MySQL database being initialized.
+
+To stop the Docker environment and free up port `80` and `3306` open a new shell and run:
+
+	docker-compose down
+
+Visit [localhost:8025](http://localhost:8025) to check all emails sent by WordPress.
+
+Add the following entry to your hosts file if you want to map `localhost` to a domain like [wp-foo-bar.local](http://wp-foo-bar.local).
+
+	127.0.0.1 wp-foo-bar.local
+
+### Scripts
+
+We use `npm` as the canonical task runner for the project. Some of the PHP related scripts are defined in `composer.json` and described further down.
+
+**Important**: The commands that execute unit tests or generate coverage reports (i.e. contain `test` in the name) should be executed inside the Docker container.
+
+#### NPM
+
+- `npm run build` to build the plugin JS and CSS assets. Use `npm run dev` to watch and re-build as you work.
+
+- `npm run lint` to lint both PHP and JS files.
+
+- `npm run lint:js` to lint only JavaScript files with [eslint](https://eslint.org/).
+
+- `npm run lint:php` to lint only PHP files with [phpcs](https://github.com/squizlabs/PHP_CodeSniffer).
+
+- `npm run test:js` to run tests for JS.
+
+- `npm run test:js:coverage` to run tests for JS with coverage reporting.
+
+- `npm run docker -- npm run test` to run both PHP and JS tests without coverage reporting.
+
+- `npm run docker -- npm run test-with-coverage` to run both PHP and JS tests with coverage reporting.
+
+- `npm run docker -- npm run test:php` to run tests for PHP.
+
+- `npm run docker -- npm run test:php:coverage` to run tests for PHP with coverage reporting.
+
+
+#### Composer
+
+There are several Composer scripts available that you also have access to — these are _mostly_ all used in `npm` scripts above.
+
+- `composer format` to format the PHP files with [phpcbf](https://github.com/squizlabs/PHP_CodeSniffer).
+
+- `composer lint` to lint the PHP files with [phpcs](https://github.com/squizlabs/PHP_CodeSniffer).
+
+- `composer readme` to generate the `readme.md` from the `readme.txt`.
+
+    _The `readme.md` file will automatically generate on commit when the `readme.txt` changes_
+
+- `composer test` to run PHPUnit tests without generating a coverage report.
```

Comment:
@derekherman `composer test` and `composer test-coverage` command fails if not run in Docker. Maybe we should delete these command?

Reply:
They are supposed to be ran in Docker

Reply:
We can add another callout though to say this needs to be ran in the container

Reply:
I am thinking that the composer commands are kind of redundant from the npm commands except for the readme command (could maybe done with npm too). But these are minor for our purpose right now i guess

Reply:
I'll create `npm` scripts and remove the `composer` ones so there is less confusion.